### ISSUE TITLE
Need hot links for website

### ISSUE THREAD

**Issue by geekedtv** [SergiX44/XBackBone#357]
I need the ability to be able to hotlink/embed my images for websites. Direct links still take you to the xbackbone page and images that are copied with the /raw appendix are not seen as images when copying from clipboard. How can I fix this?

Use case; embedding images into the editor on WordPress and Bookstack do not work.

**Comment by SergiX44** [comment]
Copying the raw link should be enough, can you try to add the image extension. Like https://example.com/xxx/yyy/raw.png

**Comment by geekedtv** [comment]
> 
> 
> Copying the raw link should be enough, can you try to add the image extension. Like https://example.com/xxx/yyy/raw.png

I tried this and the image is still broken in the editor. Weird.
Cloudflare is not blocking hot linking either. I don't know what else it could be. Something about the link structure it doesn't like.

I have tried ShareX with Chevereto and it works fine. But I like this app better for an all in one solution but if I can't get image embeds to work on websites I can't use it the way I intend to.

**Comment by geekedtv** [comment]
It appears to be working. I have to manually take the raw URL every time and delete the .png then add /raw.png?

Is there a way to automate this or edit a file so it does this for me automatically?

**Comment by geekedtv** [comment]
> 
> 
> What I suppose is that the wp editor doesn't recognize the raw image because it doesn't have an extension.
> If you go to your profile you can enable as "copy mode" the raw url instead of the preview url.

I did this already but that doesn't add the .png --  it just outputs https://example.com/xxx/yyy.png/raw I need it to be https://example.com/xxx/yyy/raw.png -- it appears there is a mistake in the url output maybe in the code. If not then the docker image I am using? https://hub.docker.com/r/linuxserver/xbackbone

Here is a video to show you what the links are doing in my address bar when I switch to raw view https://snip.lol/WeXO3/KAhawOBe08.mp4

**Comment by SergiX44** [comment]
> > What I suppose is that the wp editor doesn't recognize the raw image because it doesn't have an extension.
> > If you go to your profile you can enable as "copy mode" the raw url instead of the preview url.
> 
> I did this already but that doesn't add the .png -- it just outputs https://example.com/xxx/yyy.png/raw I need it to be https://example.com/xxx/yyy/raw.png -- it appears there is a mistake in the url output maybe in the code. If not then the docker image I am using? https://hub.docker.com/r/linuxserver/xbackbone
> 
> Here is a video to show you what the links are doing in my address bar when I switch to raw view https://snip.lol/WeXO3/KAhawOBe08.mp4
> 
> So the question is, what file can i edit to change the way it outputs the raw links?

I know the extension is not added. This is where the copy link is placed: https://github.com/SergiX44/XBackBone/blob/master/resources/templates/upload/public.twig#L34

**Comment by geekedtv** [comment]
I can't go as far to edit the file. I see where it says "raw" but the extension itself needs to be replaced with .raw. Where in the file or what lines?

**Comment by SergiX44** [comment]
> I can't go as far to edit the file. I see where it says "raw" but the extension itself needs to be replaced with .raw. Where in the file or what lines? I'd be a happy monitary supporter if this can be added in the next release!

Next release will include this improvement

**Comment by geekedtv** [comment]
> 
> 
> > I can't go as far to edit the file. I see where it says "raw" but the extension itself needs to be replaced with .raw. Where in the file or what lines? I'd be a happy monitary supporter if this can be added in the next release!
> 
> Next release will include this improvement

Very good. Should we keep this open so I know when the update is made?

Thank you!

**Comment by SergiX44** [comment]
Added in the 3.4.1 release.

**Comment by geekedtv** [comment]
I see the "embed images" toggle. I turned it on but nothing has changed?

**Comment by SergiX44** [comment]
> I see the "embed images" toggle. I turned it on . It appears to work but there are 2 extensions in the URL like this
> 
> KuqeVoCO64.png/raw.png

Yep i know, I'm going to rework a bit how urls are made anyway, next version should be fixed that minor thing.
I'm reopening this for tracking purposes.
